vit-servicing-station-lib/src/v0/endpoints/snapshot/handlers.rs

```diff
@@ -47,23 +42,18 @@ pub async fn get_voters_info(
                 )
             }
         }
-        Ok(None) => Err(warp::reject::not_found()),
-        Err(_) => Ok(
-            warp::reply::with_status("Database error", StatusCode::INTERNAL_SERVER_ERROR)
-                .into_response(),
-        ),
+        Err(err) => Ok(err.into_response()),
     }
 }
 
 #[tracing::instrument(skip(context))]
 pub async fn get_tags(context: SharedContext) -> Result<impl Reply, Rejection> {
-    match context.get_tags().map(|tags| warp::reply::json(&tags)) {
+    match super::get_tags(context)
+        .await
+        .map(|tags| warp::reply::json(&tags))
+    {
```

Comment:
Slight style nit: can we pull this expression into a variable, and the have `match tags {` ?